torch/quantization/convert_modules.py

```diff
@@ -0,0 +1,244 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch.nn as nn
+import torch.nn.quantized as nnq
+import torch
+
+def propagate_qconfig_helper(module, qconfig_dict, qconfig_parent=None, prefix=''):
+    r"""This is a helper function for `propagate_qconfig`
+
+    Args:
+        module: input module
+        qconfig_dict: dictionary that maps from name of submodule to quantization
+                     configuration
+        qconfig_parent: quantization config of parent module, we will fallback to
+                       this config when there is no specified config for current
+                       module
+        prefix: corresponding prefix of the current module, used as key in
+                qconfig_dict
+
+    Return:
+        None, module is modified inplace with qconfig attached
+    """
+    if not hasattr(module, 'qconfig'):
+        module.qconfig = None
+        if qconfig_dict and prefix in qconfig_dict:
+            module.qconfig = qconfig_dict[prefix]
+        else:
+            module.qconfig = qconfig_parent
+
+    for name, child in module.named_children():
+        module_prefix = prefix + '.' + name if prefix else name
+        propagate_qconfig_helper(child, qconfig_dict, module.qconfig, module_prefix)
+
+def propagate_qconfig(module, qconfig_dict=None):
+    r"""Propagate qconfig through the module hierarchy and assign `qconfig`
+    attribute on each leaf module
+
+    Args:
+        module: input module
+        qconfig_dict: dictionary that maps from name of submodule to quantization
+            configuration, qconfig applies to all submodules of a given
+            module unless qconfig for the submodules are specified(when the
+            submodule already has qconfig attribute)
+
+    Return:
+        None, module is modified inplace with qconfig attached
+    """
+    propagate_qconfig_helper(module, qconfig_dict)
+
+def _observer_forward_hook(self, input, output):
+    r"""Forward hook that calls observer on the output
+    """
+    self.observer(output)
+
+def add_observer(module):
```

Comment:
I wonder whether we should have also `remove_observer`. The thing is that right now you might attach observers to some modules that you don't swap out later. So might make sense to clean up. (for removing hooks - you can store hook inside observer class and then do .remove() on it)

Reply:
OK, I'll add a TODO for this